src/module/actor/character/crafting/ability.ts

```diff
@@ -56,9 +56,20 @@ class CraftingAbility implements CraftingAbilityData {
         if (this.isAlchemical) {
             this.resource = "infused-reagents";
             this.isDailyPrep = true;
+            this.isPrepared = true;
         }
     }
 
+    /** Returns true if this ability cannot prepare more formulas */
+    get isFull(): boolean {
+        // Prepared with a resource can prepare infinitely, we validate when we try to craft
+        if (this.isPrepared && this.resource) {
+            return false;
+        }
+
+        return !this.maxSlots || this.preparedFormulaData.length >= this.maxSlots;
+    }
```

Comment:
No a value/max object of slots?

Reply:
What would the property be called? Also we'd have to set max to infinity if its prepared with resources and that might be misleading.